src/api/routes/example.js

```diff
@@ -22,10 +23,21 @@ module.exports = (app) => {
         });
     });
 
+    const p1 = async (req, res, next) => {
+        await Promise.resolve();
+        if (req.query.p !== "1") throw new APIError(HTTPStatus.BAD_GATEWAY, ErrorTypes.FORBIDDEN, "MUST BE 1");
```

Comment:
Not understanding why we use `throw` here and in `p2` `next(new Error)`

Reply:
It was to simulate that both work and we end up with both errors being caught correctly. In earlier versions of express, throwing inside async functions would make it hang (due to unresolved promises) and the request would not even finish. Now it does, responding with unknown error or the specific APIError, depending on what is thrown.

This is useful in cases where middleware has DB calls, that can throw, but we don't want those errors to leak, but at the same time we want the request to fail, and not hang.

Reply:
Oh I see. That's great, I think I had noticed requests hanging because of async throws. Great that it's solved!